populus/api/utils.py

```diff
@@ -0,0 +1,110 @@
+import os
+import shutil
+
+from populus.utils.filesystem import (
+    ensure_path_exists,
+)
+
+from geth.wrapper import (
+    construct_test_chain_kwargs,
+)
+
+from geth.accounts import (
+    create_new_account,
+)
+
+from geth.install import (
+    chmod_plus_x,
+)
+
+
+INIT_COMMAND = (
+    "geth --rpc --rpcaddr 127.0.0.1 --rpcport 8545 "
+    "--rpcapi admin,debug,eth,miner,net,personal,shh,txpool,web3,ws "
+    "--ws --wsaddr 127.0.0.1 --wsport 8546 "
+    "--wsapi admin,debug,eth,miner,net,personal,shh,txpool,web3,ws "
+    "--datadir {data_dir} --maxpeers 0 --networkid 1234 --port 30303 "
+    "--ipcpath {ipc_path} --nodiscover --mine --minerthreads 1 "
+    "init {genesis_path}"
+)
+
+RUN_COMMAND = (
+    "geth --rpc --rpcaddr 127.0.0.1 --rpcport 8545 "
+    "--rpcapi admin,debug,eth,miner,net,personal,shh,txpool,web3,ws "
+    "--ws --wsaddr 127.0.0.1 --wsport 8546 "
+    "--wsapi admin,debug,eth,miner,net,personal,shh,txpool,web3,ws "
+    "--datadir {data_dir} --maxpeers 0 --networkid 1234 --port 30303 "
+    "--ipcpath {ipc_path} --unlock {account} "
+    "--password {password_path} --nodiscover --mine --minerthreads 1 "
+)
+
+GENESIS_BLOCK = '''
+{
+    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+  "coinbase": "%s",
+   "extraData": "0x686f727365",
+   "config": {
+       "daoForkBlock": 0,
+     "daoForSupport": true,
+    "homesteadBlock": 0
+    },
+   "timestamp": "0x0",
+    "mixhash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "nonce": "0xdeadbeefdeadbeef",
+    "alloc": {
+        "%s":{
+          "balance": "1000000000000000000000000000000"
+      }
+      },
+    "gasLimit": "0x47d5cc",
+    "difficulty": "0x01"
+}
+'''
+
+
+def new_local_chain(project_dir, chain_dir):
+    chains_path = os.path.join(project_dir, 'chains')
+    ensure_path_exists(chains_path)
+    chain_dir = os.path.join(chains_path, chain_dir)
```

Comment:
This seems weird to have `chain_dir` get overwritten here.  Maybe rename the original to `chain_name`?